test-d/partial-deep.ts

```diff
@@ -0,0 +1,32 @@
+import {expectError, expectType} from 'tsd';
+import {PartialDeep} from '..';
+
+const foo = {
+	baz: 'fred',
+	bar: {
+		object: {key: 'value'},
+		string: 'waldo',
+		number: 1,
+		boolean: false,
+		symbol: Symbol('test'),
+		null: null,
+		undefined: undefined, // eslint-disable-line object-shorthand
+		map: new Map<string, string>(),
+		set: new Set<string>(),
+		array: ['foo']
+	}
+};
```

Comment:
Please extend the tests here analogous to the tests for `ReadonlyDeep`.

Reply:
Did that, but please take a look at [my comment](https://github.com/sindresorhus/type-fest/pull/47#discussion_r312698373) at the other conversation.